### PR TITLE
Bugfix sso provider

### PR DESCRIPTION
**Pull request by SamRemis**
*Issue #, if available:*
#2668
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by stobrien89 — CHANGES_REQUESTED** [review]
Small change requested and a few comments.  LGTM otherwise

File: src/Credentials/CredentialProvider.php [hunk 1]
```diff
@@ -921,22 +847,131 @@ public static function shouldUseEcs()
...
      * @param array $ssoProfile
+     * @param string $clientRegion
      * @param string $accessToken
      * @param array $config
      * @return array|null
      */
-    private static function getCredentialsFromSsoService($ssoProfile, $accessToken, $config)
+    private static function getCredentialsFromSsoService($ssoProfile, $clientRegion, $accessToken, $config)
     {
         if (empty($config['ssoClient'])) {
```
[the remark this answers is not in the document]
Comment (SamRemis) [reply src/Credentials/CredentialProvider.php:972]: When the client is empty?  It's the default for both workflows.  Most customers will not purposely be explicitly be passing a client, but in the case that they want to specify client config parameters on the client we use, that's how they would do it.

File: src/Token/SsoTokenProvider.php [hunk 2]
```diff
@@ -158,7 +158,7 @@ public static function getTokenLocation($sso_session)
     {
         return self::getHomeDir()
             . '/.aws/sso/cache/'
-            . utf8_encode(sha1($sso_session))
+            . sha1($sso_session)
```
[the remark this answers is not in the document]
Comment (SamRemis) [reply src/Token/SsoTokenProvider.php:161]: Need to look into this a bit more, but I believe that sha1 will always return a utf8 compatible string and no encoding will be done.

File: src/Credentials/CredentialProvider.php [hunk 3]
```diff
@@ -921,22 +847,131 @@ public static function shouldUseEcs()
...
+            )
+        );
     }
 
+    /**
+     * @param $profiles
+     * @param $ssoProfileName
+     * @param $filename
+     * @param $config
+     * @return Promise\FulfilledPromise|Promise\Promise|Promise\PromiseInterface|Promise\RejectedPromise
```
[the remark this answers is not in the document]
Comment (SamRemis) [reply src/Credentials/CredentialProvider.php:910]: Oh yeah it does; it was autogenerated by the refactoring.  I will edit this manually.

**Review by stobrien89 — APPROVED** [review]
:shipit:

**Pull request merged**